### DISCUSSION TITLE
Connection to graph  is not working

### DISCUSSION THREAD

**Discussion by shanvelc** [aws/graph-explorer#75] [Q&A]
I have followed the installation guide and was able to install the graph explorer. But unable to connect to graph DB. Could you please provide the details what i am missing.Thanks.

**Comment by triggan** [comment]
Can you please provide further details about your configuration? What graph database are you attempting to connect to?  Where are you deploying graph explorer? Etc.

**Comment by sanchit231** [comment]
I am also having connectivity issue. Deployed graph explorer on EC2 machine and getting synchronization issue with Neptune DB. Neptune is not publically accessible

**Comment by shanvelc** [comment]
Now we are able to connect. But data is not getting viewed when we create a RDF manually. Instead if we use SPARQL and insert data it works. What would be the issue?

**Comment by shanvelc** [comment]
> Can you please provide further details about your configuration? What graph database are you attempting to connect to? Where are you deploying graph explorer? Etc.

Check if the port of 8182 is open in Neptune SG for EC2 proxy to access.

**Comment by triggan** [comment]
> Now we are able to connect. But data is not getting viewed when we create a RDF manually. Instead if we use SPARQL and insert data it works. What would be the issue?

Not sure I follow what you mean by "create a RDF manually".  Can you please elaborate?

In Neptune, Property Graph and RDF data are stored separately.  So if you write data using SPARQL/RDF, you can only access that data using SPARQL in RDF format.  Subsequently, if you write data using Gremlin or openCypher (both Property Graph languages) you can only then read that data using Gremlin or openCypher.

**Comment by sanchit231** [comment]
> > Can you please provide further details about your configuration? What graph database are you attempting to connect to? Where are you deploying graph explorer? Etc.
> 
> Check if the port of 8182 is open in Neptune SG for EC2 proxy to access.

Yes it is open when I try to telnet neptune from EC2 machine. Its working fine on 8182 port

**Comment by triggan** [comment]
@sanchit231 - Are you still experiencing issues?  What error messages are you seeing (either from the docker logs or from your browser's Dev Tools console)?

**Reply by pedrodcsjostrom** [reply]
From my EC2 I'm reaching neptune endpoint, I even can do queries from the console using curl. But when i add the connection in the graph-explorer's UI it times out and the error says "Error connecting to db"

**Comment by liqun-dodo** [comment]
I have a connectivity issue with graph-explorer as well. I have graph-explorer installed on ec2, and used command line "sudo docker run -p 443:443 --env HOST=<ec2-ip> graph-explorer" to start. I added a new connection from the graph-explorer UI: Graph Type
PG (Property Graph), Public or Proxy Endpoint: https://<neptune-endpoint>:8182 without a proxy (see attached). I got "Unable to connect with the Database" error. I noticed that the log message shows "Master credentials available Proxy server located at https://localhost" when I started the graph-explorer docker image. Not sure this "localhost" is correct. 
BTW, I am able to access this neptune endpoint from ec2 and my local laptop directly via gremlinpython lib; and also able to ping "https://<neptune-endpoint>:8182/status" from a browser. What am i doing wrong? Thanks!

<img width="663" alt="Screen Shot 2023-04-11 at 9 53 10 AM" src="https://user-images.githubusercontent.com/130474403/231188438-2ae13ee0-ae8a-4117-986d-74d7ce6f9185.png">

**Reply by pedrodcsjostrom** [reply]
Same situation here, from the EC2 im able to hit neptune from the console... even do some REST queries. But i get the same "Error connecting to the database", did you figure it out??

**Comment by GenYume** [comment]
Hello, I'm having the same issue as the other people here.

I've loaded the twitter sample dataset into Neptune, but can't seem to connect to the instance.

I've got an err NAME_NOT_FOUND when trying to connect to the database.

Has anyone been able to troubleshoot this ?
